src/firewall/core/rich.py

```diff
@@ -236,8 +278,7 @@ def check(self, family=None):
                 "nflog 'queue-size' must be an integer between 0 and 65535.",
             )
 
-        if self.limit is not None:
-            self.limit.check(family=family)
+        super().check(family=family)
 
 
 class Rich_Audit(_Rich_Entry):
```

Comment:
`audit` is a type of log. This could derive `_Rich_Log`.

Reply:
I chose the base types based on how the entries are assigned to the rich_rule.

For example,
- `rich_rule.audit` property holds  instance derived from `_Rich_Audit`
- `rich_rule.log` property holds instance derived from `_Rich_Log`
- `rich_rule.action` property holds instance derived from `_Rich_Action`

Under that consideration, `Rich_Audit` "is not a" `_Rich_Log`.

Instead, I added `_Rich_EntryWithLimit`, which is now a base class of `Rich_Audit`, `_Rich_Action` and `_Rich_Log`.